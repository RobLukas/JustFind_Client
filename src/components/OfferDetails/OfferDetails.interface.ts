import TechnologyType from '@api/technologyCollection';

interface OfferDetailsProps {
  mainlyTechnology: TechnologyType;
  logoSrc: string;
  title: string;
  salary: {
    from: number;
    to: number;
    currency: string;
  };
  companyName: string;
  companyAddress: string;
  technologies: string[];
  isNew: boolean;
  addedDate: number;
}

export default OfferDetailsProps;
