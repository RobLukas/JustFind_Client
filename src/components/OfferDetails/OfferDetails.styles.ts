import styled from 'styled-components';

const OfferDetails = styled.div(({ theme }) => ({
  '@media (max-width: 1020px)': {
    height: 86,
  },
  height: 77,
  cursor: 'pointer',
  backgroundColor: theme.styles.secondary.background,
  borderRadius: 6,
  margin: '0 10px 12px 10px',
  padding: 0,
  display: 'flex',
}));

const OfferDetailsWrapper = styled.div({
  width: '100%',
  display: 'flex',
});

const LogoCompanyWrapper = styled.div({
  flex: '0 0 112px',
  position: 'relative',
});

const LinkWrapper = styled.a`
  display: flex;
  text-decoration: none;
  min-width: 0px;
  height: 100%;
  font-size: 11px;
  align-items: center;
  flex: 1;
`;

const DescriptionWrapper = styled.div({
  width: '100%',
  paddingRight: 20,
  minWidth: 0,
});

const TitleWrapper = styled.div({
  '@media (max-width: 1020px)': {
    display: 'block',
  },
  display: 'flex',
  alignItems: 'center',
});

const SalaryWrapper = styled.div({
  marginLeft: 'auto',
  display: 'flex',
  alignItems: 'center',
});

const NewBar = styled.span(({ theme }) => ({
  '@media (max-width: 1020px)': {
    marginLeft: 'auto',
  },
  color: theme.styles.offers.newTitle.color,
  backgroundColor: theme.styles.offers.newTitle.background,
  padding: '4px 7px',
  borderRadius: 16,
  marginLeft: 20,
  whiteSpace: 'nowrap',
  fontWeight: 400,
  verticalAlign: 'bottom',
  fontSize: 12,
}));

const AddressWrapper = styled.div({
  '@media (max-width: 1020px)': {
    display: 'none',
  },
  display: 'flex',
  alignItems: 'center',
  marginTop: 6,
});

const SeparateCompanyInfo = styled.div`
  margin-left: 20px;
`;

const SeparateDateAgoFromSalary = styled.div`
  margin-right: 21px;
`;

const TechnologyWrapper = styled.div`
  margin-left: auto;
  white-space: nowrap;
  flex: 1;
  display: flex;
  justify-content: flex-end;
  overflow: hidden;
`;

export default {
  OfferDetails,
  OfferDetailsWrapper,
  LogoCompanyWrapper,
  LinkWrapper,
  DescriptionWrapper,
  TitleWrapper,
  SalaryWrapper,
  NewBar,
  AddressWrapper,
  SeparateCompanyInfo,
  TechnologyWrapper,
  SeparateDateAgoFromSalary,
};
