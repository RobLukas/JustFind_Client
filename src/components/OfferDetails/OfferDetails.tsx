import React, { FC } from 'react';
import { v4 as uuid } from 'uuid';

import OfferDetailsProps from './OfferDetails.interface';
import LinearBackground from './LinearBackground/LinearBackground';
import LogoCompany from './LogoCompany/LogoCompany';
import Salary from './Salary/Salary';
import S from './OfferDetails.styles';
import Title from './Title/Title';
import CompanyAddressInfo from './CompanyAddressInfo/CompanyAddressInfo';
import { Business, Place } from '@material-ui/icons';
import TileRoundedEdges from './TileRoundedEdges/TileRoundedEdges';

const OfferDetails: FC<OfferDetailsProps> = ({
  mainlyTechnology,
  logoSrc,
  title,
  salary,
  companyName,
  companyAddress,
  technologies,
  isNew,
  addedDate,
}: OfferDetailsProps) => {
  return (
    <S.OfferDetails>
      <S.OfferDetailsWrapper>
        <LinearBackground technology={mainlyTechnology} />
        <S.LinkWrapper>
          <S.LogoCompanyWrapper>
            <LogoCompany logoSrc={logoSrc} />
          </S.LogoCompanyWrapper>
          <S.DescriptionWrapper>
            <S.TitleWrapper>
              <Title>{title}</Title>
              <S.SalaryWrapper>
                <Salary salary={salary} />
                <S.SeparateDateAgoFromSalary />
                {isNew ? (
                  <S.NewBar>New</S.NewBar>
                ) : (
                  <TileRoundedEdges>{`${addedDate} ago`}</TileRoundedEdges>
                )}
              </S.SalaryWrapper>
            </S.TitleWrapper>
            <S.AddressWrapper>
              <CompanyAddressInfo icon={Business}>
                {companyName}
              </CompanyAddressInfo>
              <S.SeparateCompanyInfo />
              <CompanyAddressInfo icon={Place}>
                {companyAddress}
              </CompanyAddressInfo>
              <S.TechnologyWrapper>
                {technologies.map((technology) => (
                  <TileRoundedEdges key={uuid()}>{technology}</TileRoundedEdges>
                ))}
              </S.TechnologyWrapper>
            </S.AddressWrapper>
          </S.DescriptionWrapper>
        </S.LinkWrapper>
      </S.OfferDetailsWrapper>
    </S.OfferDetails>
  );
};

export default OfferDetails;
