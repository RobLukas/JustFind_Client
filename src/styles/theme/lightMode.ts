export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
