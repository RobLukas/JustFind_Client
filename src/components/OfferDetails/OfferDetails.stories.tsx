import React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';

import OfferDetails from './OfferDetails';
import OfferDetailsProps from './OfferDetails.interface';
import ThemeWrapper from '../../containers/ThemeWrapper/ThemeWrapper';

export default {
  title: 'component/OfferDetails',
  component: OfferDetails,
} as Meta;

export const Template: Story<OfferDetailsProps> = (args: OfferDetailsProps) => (
  <ThemeWrapper>
    <OfferDetails {...args} />
  </ThemeWrapper>
);

export const jsNewOfferTemplate = Template.bind({});
jsNewOfferTemplate.args = {
  mainlyTechnology: 'JS',
  logoSrc:
    'https://bucket.justjoin.it/offers/company_logos/thumb/b4a0837f135fe43ab93209dbd8e4ce660cbd39b0.jpg?1576668518',
  title: 'Programista Javascript',
  salary: {
    from: 1000,
    to: 10000,
    currency: 'PLN',
  },
  companyName: 'Sii',
  companyAddress: 'Grunwaldzka 20/3',
  technologies: ['react', 'es6', 'css'],
  createdAt: new Date(new Date().setDate(new Date().getDate() - 1)),
};

export const htmlOfferTemplate = Template.bind({});
htmlOfferTemplate.args = {
  mainlyTechnology: 'Html',
  logoSrc:
    'https://bucket.justjoin.it/offers/company_logos/thumb/b4a0837f135fe43ab93209dbd8e4ce660cbd39b0.jpg?1576668518',
  title: 'Programista Html',
  salary: {
    from: 4000,
    to: 12500,
    currency: 'PLN',
  },
  companyName: 'Goyello',
  companyAddress: 'Łódźka 2/13a',
  technologies: ['css', 'html5', 'web design'],
  createdAt: new Date(new Date().setDate(new Date().getDate() - 8)),
};
