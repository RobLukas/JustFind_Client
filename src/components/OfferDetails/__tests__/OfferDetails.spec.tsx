import lightMode from 'styles/theme/lightMode';
import { cleanup, render } from '@testing-library/react';
import 'jest-styled-components';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import OfferDetails from '../OfferDetails';
import OfferDetailsProps from '../OfferDetails.interface';

describe('<OfferDetails />', () => {
  afterEach(cleanup);
  it('render correctly', () => {
    const props: OfferDetailsProps = {
      mainlyTechnology: 'JS',
      logoSrc: 'logoSrc',
      title: 'Job Title',
      salary: {
        from: 1000,
        to: 2000,
        currency: 'PLN',
      },
      companyName: 'company',
      companyAddress: 'address',
      technologies: ['html', 'css', 'js'],
      createdAt: new Date(),
    };

    const { container } = render(
      <ThemeProvider theme={lightMode}>
        <OfferDetails {...props} />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
