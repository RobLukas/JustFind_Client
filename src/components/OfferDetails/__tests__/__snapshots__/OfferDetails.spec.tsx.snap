// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OfferDetails /> render correctly 1`] = `
<div
  class="sc-eCApGN fOsrME"
>
  <div
    class="sc-jSFkmK iAzcCZ"
  >
    <div
      class="sc-bdnylx jpNXAy"
    />
    <a
      class="sc-iCoHVE hcwDGp"
    >
      <div
        class="sc-gKAblj fXMzXJ"
      >
        <img
          class="sc-gtssRu dflBGq"
          src="logoSrc"
        />
      </div>
      <div
        class="sc-fujyUd gwQUaz"
      >
        <div
          class="sc-pNWxx flpjmn"
        >
          <span
            class="sc-bqGHjH hkDKqO"
          >
            Job Title
          </span>
          <div
            class="sc-jrsJCI eqPDCX"
          >
            <span
              class="sc-hKFyIo bEfZPG"
            >
              1000 - 2000 PLN
            </span>
            <div
              class="sc-crzoUp jOmbSh"
            />
            <span
              class="sc-fFSRdu dJJkKY"
            >
              New
            </span>
          </div>
        </div>
        <div
          class="sc-kEqYlL fACiCd"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-root sc-hBMVcZ kHjMiS"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"
            />
          </svg>
          <span
            class="sc-ksluoS czSlKN"
          >
            company
          </span>
          <div
            class="sc-iqAbSa kVdBPr"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-root sc-hBMVcZ kHjMiS"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
          <span
            class="sc-ksluoS czSlKN"
          >
            address
          </span>
          <div
            class="sc-dIsAE iUivZL"
          >
            <span
              class="sc-fnVYJo iyZRAM"
            >
              html
            </span>
            <span
              class="sc-fnVYJo iyZRAM"
            >
              css
            </span>
            <span
              class="sc-fnVYJo iyZRAM"
            >
              js
            </span>
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
